backend/typescript/rest/creatorRoutes.ts

```diff
@@ -1,11 +1,114 @@
 import { Router } from "express";
-import { creatorDtoValidator } from "../middlewares/validators/creatorValidators";
+import { isAuthorizedByRole } from "../middlewares/auth";
+import CreatorService from "../services/implementations/creatorService";
+import ICreatorService from "../services/interfaces/creatorService";
+import { sendErrorResponse } from "../utilities/errorResponse";
 
 const creatorRouter: Router = Router();
 creatorRouter.get("/");
 
-creatorRouter.get("/", creatorDtoValidator, async (req, res) => {
-  res.status(200).send("hello creator");
-});
+const creatorService: ICreatorService = new CreatorService();
+
+function isOverlap(ageRange: string, ageRangeCmp: string): boolean {
+  const lowerInclusiveBound = parseInt(
+    ageRange.split(",")[0].slice(1) || "",
+    10,
+  );
+  const upperExclusiveBound = parseInt(
+    ageRange.split(",")[1].slice(0, -1) || "",
+    10,
+  );
+
+  const lowerInclusiveBoundCmp = parseInt(
+    ageRangeCmp?.split(",")[0].slice(1) || "",
+    10,
+  );
+  const upperExclusiveBoundCmp = parseInt(
+    ageRangeCmp?.split(",")[1].slice(0, -1) || "",
+    10,
+  );
+
+  return (
+    Math.max(lowerInclusiveBound, lowerInclusiveBoundCmp) <=
+    Math.min(upperExclusiveBound, upperExclusiveBoundCmp)
+  );
+}
+
+creatorRouter.get(
+  "/",
+  isAuthorizedByRole(new Set(["Admin", "Subscriber", "Author"])),
+  async (req, res) => {
+    const { id, location, ageRange, genre, status } = req.query;
+    const isAdmin = !isAuthorizedByRole(new Set(["Admin"]));
+    if (id) {
+      if (typeof id === "string") {
+        const idNumeric = parseInt(id, 10);
+        // Get User By Id
+        if (Number.isNaN(idNumeric)) {
+          res
+            .status(400)
+            .json({ error: "id query parameter must be a number." });
+        } else {
+          try {
+            const creator = await creatorService.getCreatorById(id);
+            if (isAdmin && creator.isApproved === false) {
```

Comment:
This filtering logic, checking if the creator is approved, should happen in the service imo